app/src/androidTest/java/com/github/ybecker/epforuml/latex/LatexDialogTest.kt

```diff
@@ -0,0 +1,217 @@
+
+        onView(ViewMatchers.withId(R.id.latex_input))
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+        onView(ViewMatchers.withId(R.id.latex_editText))
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+        onView(ViewMatchers.withId(R.id.latex_render_button))
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+        onView(ViewMatchers.withId(R.id.latex_output))
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+        onView(ViewMatchers.withId(R.id.latex_mathView))
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+        onView(ViewMatchers.withId(R.id.latex_save_button))
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+        onView(ViewMatchers.withId(R.id.latex_cancel_button))
+            .check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+    }
+
+    @Test
+    fun checkTextsAreEmptyWhenTargetEditTextIsEmpty() {
+        lateinit var dialog: LatexDialog
+        lateinit var targetEditText: EditText
+        scenario.onActivity {
+            targetEditText = EditText(it)
+            targetEditText.setText("")
+            dialog = LatexDialog(it, targetEditText)
+            dialog.show()
+        }
+        val input = dialog.findViewById<EditText>(R.id.latex_editText)
+        assertThat(input.text.toString(), Is(""))
+
+        val rendered = dialog.findViewById<MathView>(R.id.latex_mathView)
+        assertThat(rendered.getDisplayText(), Is(""))
+    }
+
+    @Test
+    fun checkTextsAreNotEmptyWhenTargetEditTextIsNot() {
+        lateinit var dialog: LatexDialog
+        lateinit var targetEditText: EditText
+        val text = "test"
+        scenario.onActivity {
+            targetEditText = EditText(it)
+            targetEditText.setText(text)
+            dialog = LatexDialog(it, targetEditText)
+            dialog.show()
+        }
+        val input = dialog.findViewById<EditText>(R.id.latex_editText)
+        assertThat(input.text.toString(), Is(text))
+
+        val rendered = dialog.findViewById<MathView>(R.id.latex_mathView)
+        assertThat(rendered.getDisplayText(), Is(text))
+    }
+
+    @Test
+    fun checkMathViewChangesWhenClickOnRender() {
+        lateinit var dialog: LatexDialog
+        lateinit var targetEditText: EditText
+        val text = "test"
+        scenario.onActivity {
+            targetEditText = EditText(it)
+            targetEditText.setText(text)
+            dialog = LatexDialog(it, targetEditText)
+            dialog.show()
+        }
+        val input = dialog.findViewById<EditText>(R.id.latex_editText)
+        assertThat(input.text.toString(), Is(text))
+
+        val rendered = dialog.findViewById<MathView>(R.id.latex_mathView)
+        assertThat(rendered.getDisplayText(), Is(text))
+
+        val text2 = "test2"
+
+        Thread.sleep(1000)
+
+        onView(ViewMatchers.withId(R.id.latex_editText))
```

Comment:
Maybe you could do a function for this because you are repeating the same 10 lines of codes 3 times if you take also into account the commented test